packages/api-client/index.ts

```diff
@@ -11,13 +20,94 @@ class APIClient {
     index: async (): Promise<GetClubResponse> => {
       return (await this.axios.get("/api/club")).data;
     },
-    create: async (p: CreateClubParams): Promise<CreateClubResponse> => {
-      return (await this.axios.post("/api/club", p)).data;
+    create: async (params: CreateClubParams): Promise<CreateClubResponse> => {
+      return (await this.axios.post("/api/club", params)).data;
+    },
+  };
+  profile = {
+    index: async (): Promise<GetProfileResponse> => {
+      return (await this.axios.get(`/v1/profile`)).data;
+    },
+  };
+  course = {
+    get: async (courseId: number): Promise<GetCourseResponse> => {
+      const course = (await this.axios.get(`/api/v1/courses/${courseId}`)).data;
+      course.officeHours.forEach((officeHour: any) =>
+        parseOfficeHourDates(officeHour)
+      );
+      return course;
+    },
+  };
+  taStatus = {
+    update: async (courseId: number): Promise<TAUpdateStatusResponse> => {
+      const queue = (
+        await this.axios.patch(`/v1/courses/${courseId}/ta/change_status`)
+      ).data;
+      parseQueueDates(queue);
+      queue.questions.forEach((question: any) => parseQuestionDates(question));
+      return queue;
+    },
+  };
+  questions = {
+    index: async (queueId: number): Promise<ListQuestionsResponse> => {
+      const questions = (
+        await this.axios.get(`/v1/queues/${queueId}/questions`)
+      ).data;
+      questions.forEach((question: any) => parseQuestionDates(question));
+      return questions;
+    },
+    create: async (
+      queueId: number,
+      params: CreateQuestionParams
+    ): Promise<CreateQuestionResponse> => {
+      const question = (
+        await this.axios.post(`/v1/queues/${queueId}/questions`, params)
+      ).data;
+      parseQuestionDates(question);
+      return question;
+    },
+    get: async (
+      queueId: number,
+      questionId: number
+    ): Promise<GetQuestionResponse> => {
+      return (
+        await this.axios.get(`/v1/queues/${queueId}/questions/${questionId}`)
+      ).data;
+    },
+    update: async (
+      queueId: number,
+      questionId: number,
+      params: UpdateQuestionParams
+    ): Promise<UpdateQuestionResponse> => {
+      const question = (
+        await this.axios.patch(
+          `/v1/queues/${queueId}/questions/${questionId}`,
+          params
+        )
+      ).data;
+      parseQuestionDates(question);
+      return question;
     },
   };
   constructor(baseURL: string = "") {
     this.axios = Axios.create({ baseURL: baseURL });
   }
 }
 
+function parseOfficeHourDates(officeHour: any): void {
```

Comment:
should these be explicitly `any`? I'm fine with either defining a type for it, or keeping it implicit

Reply:
The reason it's an `any` is that its super similar to the the type `OfficeHourBlock` but the dates are strings instead of Date object. We figured since there is Joi validations on the response from the server we can assume that data will be right form. 
Now that I think of it though, I'm not sure what would happen if the server responds with an error?